src/__tests__/opportunities.int.test.ts

```diff
@@ -59,4 +59,64 @@ describe('/opportunities', () => {
       });
     });
   });
+
+  describe('/id', () => {
+    it('returns exactly one opportunity selected by id', async () => {
+      const opportunityIds: Result<{ id: number; title: string }> = await db.query(`
+        INSERT INTO opportunities (
+          title,
+          created_at
+        )
+        VALUES
+          ( '🔥', '2525-01-02T00:00:01Z' ),
+          ( '✨', '2525-01-02T00:00:02Z' ),
+          ( '🚀', '2525-01-02T00:00:03Z' )
+        RETURNING id, title;
+      `);
+      logger.debug('opportunityIds: %o', opportunityIds);
```

Comment:
I wouldn't have thought about not having them until now. What is the reason? Avoiding side effects? Keeping them reasonably clean?

Reply:
Yeah I guess cleanliness plus the idea that we don't want to clutter test suite output with logs / logs make it harder to see failing tests.

Reply:
Yes, if we need to print statements during tests, we should use the same tools as production.

As for the purpose of printing what came back from the database here, it is to help debug what went wrong when the test fails: was it because the DB sent back an unexpected thing or was it because the subsequent filter or assertion was not set up correctly?

In general, adding these statements and keeping them means you don't have to go and add a log statement or mess around with a debugger, you simply run with debug level and see what happened in more detail. If this is too verbose because the tests run at debug level by default, we could change that to info level.

If these statements in particular are annoying, they can be demoted to trace level or removed.

Does any of that make sense?

Do you usually run with a debugger? Do you usually remove all your log statements before committing?

Reply:
Totally makes sense

Re do I remove: if they're one-off statements that I used to create the thing the first time or debug an ad hoc issue I remove them.

If they're statements that are fundamental to the continuing evaluation of the health of the project I keep them as you described.

I was mostly asking because it would be good to have a consistent practice around where to add logs when it comes to tests -- there are plenty of tests that make DB calls, so maybe we should make sure to log the result in all cases?  More generally I like the idea of NOT logging manually and just configuring tinypg to do a logger.trace as part of an `onResult` hook (see the [hooks documentation](https://github.com/dynajoe/tinypg#hooks)).

I think logs it's important for logs to be silent during testing in a CI / CD environment (and I personally keep them silent locally) so the test output is unfettered but as you said that's why log levels are configurable!

Reply:
(bottom line this is resolved -- keeping it! -- we might consider that universal change another day though if you like the sound of it)